src/task.c

```diff
@@ -728,16 +728,16 @@ void NORETURN throw_internal(jl_value_t *e)
     }
     else {
         if (jl_current_task == jl_root_task) {
-            JL_PRINTF(JL_STDERR, "fatal: error thrown and no exception handler available.\n");
+            jl_printf(JL_STDERR, "fatal: error thrown and no exception handler available.\n");
             jl_static_show(JL_STDERR, e);
-            JL_PRINTF(JL_STDERR, "\n");
+            jl_printf(JL_STDERR, "\n");
             jl_exit(1);
         }
         jl_current_task->exception = e;
         finish_task(jl_current_task, e);
         assert(0);
     }
-    jl_exit(1);
+    assert(0);
```

Comment:
finish_task() and jl_longjmp() are both NORETURN, so the compiler should see that both assert(0)s are unreachable.
assert(0) is a better way to tell the human reader of the code that "this line is not reached".
If I was cleaning up this function a bit more I'd probably remove flatten out the else clause and put another assert(0) after jl_longjump().

BTW, I did not realise that Julia disables assert() in release builds. My preference is to always ship with assert() enabled (I usually implement some kind of user-friendly assert failure message / logging system).